src/gui/window/stats.rs

```diff
@@ -163,8 +171,67 @@ impl GuiWasabiWindow {
                                 ));
                             });
                         }
+                        Statistics::Nps => {
+                            ui.horizontal(|ui| {
+                                ui.monospace("NPS:");
+                                ui.with_layout(
+                                    egui::Layout::right_to_left(egui::Align::Center),
+                                    |ui| {
+                                        self.nps.tick(note_stats.passed_notes.unwrap_or(0) as i64);
+                                        ui.monospace(format!("{}", self.nps.read()));
+                                    },
+                                );
+                            });
+                        }
+                        Statistics::Polyphony => {
+                            if let Some(poly) = stats.polyphony {
+                                ui.horizontal(|ui| {
+                                    ui.monospace("Polyphony:");
+                                    ui.with_layout(
+                                        egui::Layout::right_to_left(egui::Align::Center),
+                                        |ui| {
+                                            ui.monospace(format!("{}", poly));
+                                        },
+                                    );
+                                });
+                            }
+                        }
                     };
                 }
             });
     }
 }
+
+#[derive(Default)]
+pub struct NpsCounter(VecDeque<(Instant, i64)>);
```

Comment:
I prefer a field for this even if it's just a single field, more descriptive. Tuple structs are only good for very very primitive wrappers around stuff. This is more complex.